pydantic_settings/sources.py

```diff
@@ -22,6 +22,7 @@
 
 
 DotenvType = Union[Path, List[Path], Tuple[Path, ...]]
+env_file_sentinel: DotenvType = Path('')
```

Comment:
```suggestion
ENV_FILE_SENTINEL: DotenvType = Path('')
```

I generally prefer to have module-level constants be all-caps (unless they are type aliases, like the one just above this line, in which case PascalCase is fine).

Also, can we add documentation of how this sentinel works? I don't understand what the  meaning is vs. using `None` — I think it should be easy to determine that just from reading source code, and since I can't tell what it's doing by looking at the diff in this PR, I think it would be good to add a comment either in the docstring of `DotEnvSettingsSource.__init__` or right under this line defining `env_file_sentinel` explaining what the distinction is between providing the value `None` and the value `env_file_sentinel`.